inc/Editor/DataBinding/BlockBindings.php

```diff
@@ -223,7 +225,24 @@ public static function get_value( array $source_args, WP_Block|array $block, str
 			return $fallback_content;
 		}
 
-		$query_response = self::execute_queries( $block_context, $source_args, $field_name );
+		// Generate cache key based on context and source args that affect the query
```

Comment:
This targets the case where the block with the same set of queries is duplicated. It sped up the single request time by around 10-20% depending on how much had to be loaded.

Reply:
Noticed a bug in this that the tests caught. I'm trying to fix it. If I'm not able to, then I'm going to remove it from this PR.

Reply:
I'll wait to review/test this until you've figured out that bug. @chriszarate You mentioned in a prior PR https://github.com/Automattic/remote-data-blocks/pull/442 that `wp_cache_get()` handles some of the memoizing internally. Do you know if in-memory caching here will definitely or definitely not have an impact?

Reply:
It will not have an impact, except in the rare circumstances where caching is disabled at the query level (i.e., when the query sets `TTL=-1`). We probably want to address this, but we should do so in the caching middleware, not here in the calling code. I have a PR that does this using a Guzzle utility class. However, I don't think it's a big priority; disabling caching is probably more of a debugging option than a realistic production scenario.

Reply:
In trying to figure out this bug, I have come to the same conclusion. This isn't worth doing, so I'm going to revert it.